api/client/src/test/java/org/projectnessie/client/auth/oauth2/TestOAuth2Client.java

```diff
@@ -596,6 +596,7 @@ private void handleTokenEndpoint(HttpServletRequest req, HttpServletResponse res
       Object response;
       int statusCode = 200;
       String grantType = data.get("grant_type");
+      data.remove("grant_type"); // determined by the target type
```

Comment:
This is because I changed `@Value.Default` to `@Value.Derived` for the method `TokensRequestBase.getGrantType()`. Now the property is still serialized, but deserialization from Json cannot contain the field `grant_type` anymore.

Reply:
You probably want something like this on the base class

```java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "grantType")
@JsonSubTypes({
  @JsonSubTypes.Type(value = AuthorizationCodeTokensRequest.class, name = GrantType.AUTHORIZATION_CODE),
  ...
})```

and something like this on the sub types:
```java
@JsonTypeName(GrantType.AUTHORIZATION_CODE)
```

and remove the `getGrantType()` functions.

Reply:
Oh of course! That worked, thanks!